lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb

```diff
@@ -69,20 +70,21 @@ def publishing_organisation
       end
 
       def is_part_of
-        return {} unless linked_step_by_steps.any?
+        return {} unless step_by_step_schemas.any? || collection_pages("document_collections").any?
+
         {
-          "isPartOf" => step_by_step_schemas
+          "isPartOf" => document_collections + step_by_step_schemas
         }
       end
 
-      def linked_step_by_steps
+      def step_by_step_schemas
         # We could include `related_to_step_navs` eventually too, but initially
         # link to those that we render in the "step_by_step_nav_related" component
-        @step_by_steps ||= page.content_item.dig("links", "part_of_step_navs").to_a
+        @step_by_steps ||= fetch_step_by_steps
       end
 
-      def step_by_step_schemas
-        linked_step_by_steps.map do |step_by_step|
+      def fetch_step_by_steps
```

Comment:
I'd rename this to `fetch_step_by_step_schemas` so it's more consistent with `step_by_step_schemas` which makes it slightly easier for someone to grasp that the two methods are highly related